Framework/PythonInterface/test/python/plugins/functions/SpinDiffusionTest.py

```diff
@@ -0,0 +1,47 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.simpleapi import FunctionWrapper
+
+from numpy import corrcoef, linspace, log
+from scipy.stats import linregress
+import unittest
+
+
+class SpinDiffusionTest(unittest.TestCase):
+
+    def test_spin_diffusion_when_n_dimensions_one(self):
+        x = linspace(0.01, 100.0, 1000)
+        f = FunctionWrapper("SpinDiffusion", A=1.0, DParallel=1e3, DPerpendicular=1e-2, NDimensions=1)
+        y = f(x)
+
+        # We expect the NDimensions=1 case to follow a x^-(1/2) relationship
+        slope, *_ = linregress(log(x), log(y))
+        self.assertAlmostEqual(slope, -0.5, delta=0.01)
```

Comment:
Any reason to take a different aproach here to the other tests? i.e. Fitting a line using `linregress` rather than calling `corrcoef`

Reply:
No particular reason, other than `linregress` seemed to work well for the N=1 case but it was harder to get it to work for the N=2 and 3 cases

Reply:
OK fair enough, it seems to work fine so it's not an issue. If `corrcoef` seems more robust there might be an argument for using it instead - but I note the tests have passed on all OS on jenkins so happy to keep as is. In a way it helps to understand the relation that `corrcoef` is testing in later tests!